core/src/main/scala/org/bitcoins/core/wallet/utxo/AddressTag.scala

```diff
@@ -0,0 +1,24 @@
+package org.bitcoins.core.wallet.utxo
+
+sealed trait AddressTagType {
+  def tagType: String
+}
+
+sealed trait AddressTag {
+  def name: String
+  def tagType: String
+
+  def ==(at: AddressTag): Boolean = name == at.name && tagType == at.tagType
+  def !=(at: AddressTag): Boolean = !(this == at)
+}
+
+sealed trait AddressTagFactory[TagType <: AddressTagType] {
```

Comment:
So this will be used for Bitcoin-S supported address tags that we add in the future. The `StorageLocationTags` is an example for someone importing bitcoin-s and creating their own address tags.